enterprise_learner_portal/api/v1/views.py

```diff
@@ -55,16 +42,50 @@ def get(self, request):
             user_id=user.id,
             enterprise_customer__uuid=enterprise_customer_id,
         )
-        course_ids_for_ent_enrollments = EnterpriseCourseEnrollment.objects.filter(
+        enterprise_enrollments = EnterpriseCourseEnrollment.objects.filter(
             enterprise_customer_user=enterprise_customer_user
-        ).values_list('course_id', flat=True)
-
-        overviews = get_course_overviews(course_ids_for_ent_enrollments)
+        )
 
         data = EnterpriseCourseEnrollmentSerializer(
-            overviews,
+            enterprise_enrollments,
             many=True,
             context={'request': request},
         ).data
 
         return Response(data)
+
+    def patch(self, request):
+        """
+        Patch method for the view.
+        """
+        user = request.user
+        enterprise_customer_id = request.query_params.get('enterprise_id', None)
+        course_id = request.query_params.get('course_id', None)
+        marked_done = request.query_params.get('marked_done', None)
+        if not (enterprise_customer_id and course_id) or marked_done is None:
+            return Response(
+                {'error': 'enterprise_id, course_id, and marked_done must be provided as query parameters'},
+                status=HTTP_400_BAD_REQUEST
+            )
+
+        enterprise_customer_user = get_object_or_404(
+            EnterpriseCustomerUser,
+            user_id=user.id,
+            enterprise_customer__uuid=enterprise_customer_id,
+        )
+
+        enterprise_enrollment = get_object_or_404(
+            EnterpriseCourseEnrollment,
+            enterprise_customer_user=enterprise_customer_user,
+            course_id=course_id
+        )
+
+        enterprise_enrollment.marked_done = marked_done
+        enterprise_enrollment.save()
+
+        data = EnterpriseCourseEnrollmentSerializer(
+            enterprise_enrollment,
+            context={'request': request},
+        ).data
+
+        return Response(data)
```

Comment:
Having the response as the newly updated course enrollment serialized by `EnterpriseCourseEnrollmentSerializer` matches what the frontend will be looking for :)